## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#12fa8aa6-2239-4f2c-b94e-f55252701140) |
<a name="12fa8aa6-2239-4f2c-b94e-f55252701140"></a>
### 10536-c840413/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:31681
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ff788cc348d:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 9 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... FAILED in 1 minutes and 3 seconds
{
  "name": "aws-sdk-php",
  "function": "getBucketPolicy ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5f6942178f5e0"
  },
  "duration": "40635",
  "status": "FAIL",
  "error": "Error executing \"CreateBucket\" on \"http://minio-dev8.minio.io:31681/aws-sdk-php-5f6942178f5e0\"; AWS HTTP error: Client error: `PUT http://minio-dev8.minio.io:31681/aws-sdk-php-5f6942178f5e0` resulted in a `404 Not Found` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Mes (truncated...)\n NoSuchBucket (client): The specified bucket does not exist - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>NoSuchBucket</Code><Message>The specified bucket does not exist</Message><BucketName>aws-sdk-php-5f6942178f5e0</BucketName><Resource>/aws-sdk-php-5f6942178f5e0</Resource><RequestId>1636F2E4C748E873</RequestId><HostId>faac76fe-aeab-4be3-855a-90044632e4a8</HostId></Error>"
}
(3/15) Running aws-sdk-ruby tests ... done in 23 seconds
(4/15) Running awscli tests ... done in 2 minutes and 53 seconds
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 5 minutes and 58 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 41 seconds
(8/15) Running minio-go tests ... done in 6 minutes and 28 seconds
(9/15) Running minio-java tests ... done in 10 minutes and 44 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 37 seconds
(11/15) Running minio-py tests ... FAILED in 17 minutes and 39 seconds
{
  "name": "minio-py:test_presigned_get_object_expiry",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea",
    "object_name": "c96709ef-960c-41ee-8c17-6b619fdaa6ac",
    "presigned_get_object_url": "http://minio-dev8.minio.io:31681/minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea/c96709ef-960c-41ee-8c17-6b619fdaa6ac?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioazure%2F20200922%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200922T010445Z&X-Amz-Expires=120&X-Amz-SignedHeaders=host&X-Amz-Signature=c112a76c0c7a40f53bc45be3406d247e83617da4931e6aa8e8e5e8eb66363e45",
    "response.status": 500,
    "response.reason": "Internal Server Error",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"473\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"etag\": [\"ETag\", \"\\\"6d2040290066a0bde02c04e3a24c34d6\\\"\"], \"last-modified\": [\"Last-Modified\", \"Tue, 22 Sep 2020 01:04:45 GMT\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2020-09-22T00-05-14Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"1636F588D059E23E\"], \"x-cache\": [\"X-Cache\", \"HIT\"], \"x-cache-lookup\": [\"X-Cache-Lookup\", \"HIT\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Tue, 22 Sep 2020 01:04:45 GMT\"]}}",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>InternalError</Code><Message>We encountered an internal error, please try again.</Message><Key>c96709ef-960c-41ee-8c17-6b619fdaa6ac</Key><BucketName>minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea</BucketName><Resource>/minio-py-test-b1a4cc0b-ef45-4489-9f31-2da082ef71ea/c96709ef-960c-41ee-8c17-6b619fdaa6ac</Resource><RequestId>1636F588D059E23E</RequestId><HostId>faac76fe-aeab-4be3-855a-90044632e4a8</HostId></Error>"
  },
  "message": "InternalError: message: We encountered an internal error. Please try again.",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 147, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1349, in test_presigned_get_object_expiry\n    object_name).get_exception()\nminio.error.InternalError: InternalError: message: We encountered an internal error. Please try again.\n",
  "duration": 1098
}
(11/15) Running s3cmd tests ... done in 2 minutes and 22 seconds
(12/15) Running s3select tests ... done in 1 minutes and 7 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10536-c840413

--- a/docs/bucket/replication/README.md
+++ b/docs/bucket/replication/README.md
@@ -18,9 +18,9 @@ mc admin bucket remote add myminio/srcbucket https://accessKey:secretKey@replica
 Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'
 ```
 
-> NOTE:  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster (we do not recommend running root credentials/super admin with replication). The access credentials used at the destination requires *s3:ReplicateObject* permission.
+>  The user running the above command needs *s3:GetReplicationConfiguration* and *s3:GetBucketVersioning* permission on the source cluster. We do not recommend running root credentials/super admin with replication, instead create a dedicated user. The access credentials used at the destination requires *s3:ReplicateObject* permission.
 
-Minimal policy on source bucket should have following permissions on the source bucket.
+The *source* bucket should have following minimal permission policy:
 ```
 {
  "Version": "2012-10-17",
@@ -39,7 +39,7 @@ Minimal policy on source bucket should have following permissions on the source 
   }
 }
 ```
-The access key configured to perform replication on the target cluster should have minimal permissions as in example below.
+The access key provided for the replication *target* cluster should have these minimal permissions:
 ```
 {
  "Version": "2012-10-17",
@@ -71,7 +71,7 @@ The access key configured to perform replication on the target cluster should ha
 }
 
 ```
-Once successfully created and authorized the `mc admin bucket remote add` command generates a replication target ARN.  The command below lists all the currently authorized replication targets:
+Once successfully created and authorized, the `mc admin bucket remote add` command generates a replication target ARN.  This command lists all the currently authorized replication targets:
 ```
 mc admin bucket remote ls myminio/srcbucket --service "replication"
 Role ARN = 'arn:minio:replication:us-east-1:c5be6b16-769d-432a-9ef1-4567081f3566:destbucket'

